### ISSUE TITLE
Crackly sound, Bridge server closed/bind() failed messages

### ISSUE THREAD

**Issue by NoraCodes** [VCVRack/Rack#1762]
### Details

Operating system and version: Ubuntu 19.10
Rack version (vX.Y.Z): 1.1.6

### Summary
Rack making crackling sounds and constantly creating/destroying audio devices, along with constantly repeated:

```
[2.212 info src/bridge.cpp:366] Bridge server closed
[2.312 warn src/bridge.cpp:375] Bridge server bind() failed
```

In log file.
### Steps to reproduce

Connect an audio source to an AUDIO-8 and set the output to ALSA / Default output. CPU usage of that module goes very high and audio is crackly. In Pulse Audio Volume Control, new sound sources are constantly created and destroyed.

**Comment by NoraCodes** [comment]
Oh, looks like something else is using the port. This should be handled more gracefully:

```
❯ lsof -i -P -n | grep 12512
firefox    5903 nora   15u  IPv4  7912610      0t0  TCP 127.0.0.1:12512 (LISTEN)
chrome-gn  6137 nora   15u  IPv4  7912610      0t0  TCP 127.0.0.1:12512 (LISTEN)
```

**Comment by NoraCodes** [comment]
The plot thickens - after killing those processes, and even restarting, the problem remains.

**Comment by MarcBoule** [comment]
Hi, it might be hard to get any progress with this issue unfortunately, since Bridge is not being developed further. Rack v2 will have much better DAW integration.

> VCV Bridge was deprecated in July 2018 and is now unsupported.

https://vcvrack.com/manual/FAQ#what-was-vcv-bridge

**Comment by NoraCodes** [comment]
No, no, I'm not _using_ the bridge; I'm just using ALSA, as described in "Steps to Reproduce" above. I mentioned it because that's the only abnormal, cyclic behavior that I could find in the logs, which seems like it might correspond to the problem.

**Comment by NoraCodes** [comment]
So, having had to re-install my whole computer anyway, I set up JACK and everything works fine. This makes me pretty sure it's some kind of interaction between Rack, ALSA, and Pulse.

**Comment by MarcBoule** [comment]
Yeah, Linux audio has its quirks, at least for a non-linux head like me. Glad to see it works now :-)

**Comment by NoraCodes** [comment]
> Linux audio has its quirks

Maybe so, but this isn't something I've experienced from anything else - I do think there's likely a bug in Rack causing this. See, for instance, the GIF I posted of the ALSA plugin being reinitialized several times a second - that's abnormal.

**Comment by MarcBoule** [comment]
Ok, best leave the issue open for Andrew then.

**Comment by WesleyM77** [comment]
I'm getting the crackly sound here on windows. The log doesn't show anything abnormal except
`[114.248 warn src/bridge.cpp:362] Bridge server close() failed`
when I close the program.

**Comment by LeonFedotov** [comment]
crackling sounds on macs, not sure how to pinpoint the issue but used to have issues with graphics and fps caps limiting cpu for audio

**Comment by LOGUNIVPM** [comment]
Same here. This bug has always been there (on my main laptop). The thing troubled me for months: I couldn't get decent performance while teaching to my students. Tremendous.
I suspect there are tons of opening/closing audio streams to/from alsa or pulseaudio. Sometimes the CPU manages to handle all these calls and the audio works fine, but it lasts for a short time.
Using JACK is NOT an option for remote teaching.

**Comment by LeonFedotov** [comment]
Limiting fps in the helps a bunch :)

**Comment by LOGUNIVPM** [comment]
> Limiting fps in the helps a bunch :)

I've always used 10Hz :)
Honestly, I even edited RtAudio.cpp to remove the warning to console, cause usually printing to console takes a lot of CPU cycles. But that doesn't help either.
